packages/data-stores/src/reader/mutations/use-subscription-manager-site-delivery-frequency-mutation.ts

```diff
@@ -0,0 +1,69 @@
+import { translate } from 'i18n-calypso';
+import { useMutation, useQueryClient } from 'react-query';
+import { callApi } from '../helpers';
+import { useIsLoggedIn } from '../hooks';
+import type { SiteSubscriptionDeliveryFrequency } from '../types';
+
+type SiteSubscriptionDeliveryFrequencyParams = {
+	delivery_frequency: SiteSubscriptionDeliveryFrequency;
+	blog_id: number | string;
+};
+
+type SubscriptionResponse = {
+	blog_ID: string;
+	delivery_frequency: SiteSubscriptionDeliveryFrequency;
+	status: string;
+	ts: Date;
+};
+
+type SiteSubscriptionDeliveryFrequencyResponse = {
+	success: boolean;
+	subscribed: boolean;
+	subscription: SubscriptionResponse | null;
+};
+
+const useSubscriptionManagerSiteDeliveryFrequencyMutation = () => {
+	const isLoggedIn = useIsLoggedIn();
+	const queryClient = useQueryClient();
+	return useMutation<
+		SiteSubscriptionDeliveryFrequencyResponse,
+		Error,
+		SiteSubscriptionDeliveryFrequencyParams
+	>(
+		async ( params: SiteSubscriptionDeliveryFrequencyParams ) => {
+			if ( ! params.blog_id || ! params.delivery_frequency ) {
+				throw new Error(
+					translate( 'Something went wrong while saving the delivery frequency.', {
```

Comment:
The fact that we are translating this here means, that we will use this string to display it to a user, right?

Reply:
Would it be better to leave it to the client and return something like 'missing_params' here?

Reply:
Oh, I am not sure. I am just asking because if we do not handle this, then it will only be displayed in console. Are we planning on handling this at all? I do not think we considered this yet? Right?

Reply:
No, I put it in there in case we might want to handle it in the future.

Reply:
Maybe then, for now we could just leave a comment above the line as a reminder to translate the string if we decide to display it to the user.

Reply:
I added one. But won't it be sent for translation automatically? Or am I overassuming things? 😁

Reply:
> But won't it be sent for translation automatically?

Oh, it will. I meant that we put the comment, and we do remove the call to translate.

Reply:
AS in:
```ts
throw new Error( 'Something went wrong while saving the delivery frequency.' );
```

Reply:
Ah ofcourse :D

Reply:
You addressed in one place but forgot to here. 😅

Reply:
Haha oops. Fixed now.